<launch>

  <arg name="sensorID1" default="cam0"/>
  <arg name="sensorID2" default="cam1"/>
  <arg name="sensorID3" default="flir"/>
  <arg name="bagpath" default="/media/mwmaster/b9c5bbeb-67ca-493f-8b8a-3f9a36ac549b/polygon_04-08-2024-14-50/"/>
  <arg name="sensors_yaml" default="~/camera_yamls/birdsEyeSensorParams.yaml"/>
  <arg name="clicks_csv" default="~/catch/data.csv"/>

  <executable cmd="ros2 bag play $(var bagpath) --rate 1.0"/>

  <node pkg="imu_timer" exec="imu_pad" name='imu_pad'>
    <remap from="/image" to="/cam0/camera/image_raw" />
  </node>

  <node pkg='nv12_to_rgb8' exec='nv12decode' name='$(var sensorID1)nv12decode' namespace="$(var sensorID1)">
    <remap from="/image" to="/$(var sensorID1)/camera/image_raw" />
    <remap from="/nv12_decoded" to="/$(var sensorID1)/camera/nv12_decoded" />
  </node>

  <node pkg='nv12_to_rgb8' exec='nv12decode' name='$(var sensorID2)nv12decode' namespace="$(var sensorID2)">
    <remap from="/image" to="/$(var sensorID2)/camera/image_raw" />
    <remap from="/nv12_decoded" to="/$(var sensorID2)/camera/nv12_decoded" />
  </node>

  <node pkg="image_proc" exec="image_proc" name="$(var sensorID1)image_proc" namespace="$(var sensorID1)">
    <remap from="image" to="/$(var sensorID1)/nv12_decoded" />
    <remap from="camera_info" to="/$(var sensorID1)/camera/camera_info" />
  </node>

  <node pkg="image_proc" exec="image_proc" name="$(var sensorID2)image_proc" namespace="$(var sensorID2)">
    <remap from="image" to="/$(var sensorID2)/nv12_decoded" />
    <remap from="camera_info" to="/$(var sensorID2)/camera/camera_info" />
  </node>

  <!-- <node name="$(var sensorID3)image_proc" pkg="image_proc" exec="image_proc" namespace="$(var sensorID3)">
    <remap from="image" to="/image_raw" />
    <remap from="camera_info" to="/$(var sensorID3)/camera/camera_info" />
  </node> -->

  <node pkg="birdseye" exec="sub_node" name="$(var sensorID1)sub_node" namespace="$(var sensorID1)">
    <remap from="/image" to="/$(var sensorID1)/image_rect" />
    <remap from="/imu" to="/imu/data_timeshifted"/>
    <remap from="/gps" to="/ublox_gps_node/fix"/>
    <remap from="/gps_flag" to="/ublox_gps_node/navpvt"/>
	  <remap from="/radalt" to="/rad_altitude"/>
    <param name="sensorID" value="$(var sensorID1)"/>
  </node>

  <node pkg="birdseye" exec="sub_node" name="$(var sensorID2)sub_node" namespace="$(var sensorID2)">
    <remap from="/image" to="/$(var sensorID2)/image_rect" />
    <remap from="/imu" to="/imu/data_timeshifted"/>
    <remap from="/gps" to="/ublox_gps_node/fix"/>
    <remap from="/gps_flag" to="/ublox_gps_node/navpvt"/>
	  <remap from="/radalt" to="/rad_altitude"/>
    <param name="sensorID" value="$(var sensorID2)"/>
  </node>

  <node pkg="birdseye" exec="sub_node" name="$(var sensorID3)sub_node" namespace="$(var sensorID3)">
    <remap from="/image" to="/image_raw" />
    <remap from="/imu" to="/imu/data_timeshifted"/>
    <remap from="/gps" to="/ublox_gps_node/fix"/>
    <remap from="/gps_flag" to="/ublox_gps_node/navpvt"/>
    <remap from="/radalt" to="/rad_altitude"/>
    <param name="sensorID" value="$(var sensorID3)"/>
  </node>

  <executable cmd="rqt_graph" />

</launch>
